File: app/src/main/java/com/example/cashflow/box/box_template_fragment.java
package com.example.cashflow.box;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.cashflow.R;

public class box_template_fragment extends Fragment {
    TextView boxTitle;
    Button boxButton;
    TextView boxTextView;

    public box_template_fragment() {

    }

    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.box_fragment_template, container, false);

        LinearLayout dynamicContainer = view.findViewById(R.id.boxContainer);

        for (int i = 0; i < 3; i++) {
            // Inflaziona il layout per ogni elemento dinamico
            View dynamicView = inflater.inflate(R.layout.box_fragment_template, dynamicContainer, false);

            // Trova le viste all'interno di dynamicView
            boxTitle = new TextView(getContext());
            boxButton = new Button(getContext());
            boxTextView = new TextView(getContext());

            // Imposta i testi per le viste
            boxTitle.setText("Titolo " + (i + 1));
            boxButton.setText("Pulsante " + (i + 1));
            boxTextView.setText("Testo aggiuntivo " + (i + 1));

            // Aggiungi dynamicView al container
            dynamicContainer.addView(dynamicView);
        }

        return view;
    }

}

File: app/src/main/java/com/example/cashflow/box/box_template_fragment.kt
package com.example.cashflow.box

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.util.Log
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.GridLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.example.cashflow.DetailsActivity
import com.example.cashflow.R

import com.example.cashflow.dataClass.*
import com.example.cashflow.db.*

class box_template_fragment(private val readSQL: ReadSQL, private val writeSQL: WriteSQL) : Fragment() {

    private var noDataTextView: TextView? = null
    private var viewTemplatesBtn: Button? = null

    private fun addTemplateButtons(view: View,templates: List<TemplateTransaction>) {
        val gridLayout = view.findViewById<GridLayout>(R.id.gridLayout)
        gridLayout.setColumnCount(2)

        templates.forEachIndexed { index, template ->
            val button = Button(context).apply {
                layoutParams = GridLayout.LayoutParams().apply {
                    width = 300
                    height = 100
                    rightMargin = 20
                    leftMargin = 20
                    topMargin = 20
                    setGravity(Gravity.CENTER)
                    columnSpec = GridLayout.spec(GridLayout.UNDEFINED, 1f)
                }
                text = template.name
                setTextColor(Color.WHITE)
                setBackgroundColor(Color.parseColor("#37a63e"))
                setId(View.generateViewId())
                tag = template.id // Imposta il tag del pulsante con l'ID del template
                setOnClickListener { v ->
                    val id = v.tag as Int
                    Log.d("Button", "Template $id clicked")
                }
            }
            gridLayout.addView(button)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Infla il layout per questo fragment
        return inflater.inflate(R.layout.box_fragment_template, container, false)

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        noDataTextView = view.findViewById(R.id.noDataTextView)
        viewTemplatesBtn = view.findViewById(R.id.viewTemplatesBtn)

        val templates = readSQL.getAllTemplateTransactions()
        if (templates.isEmpty() ) {
            noDataTextView?.visibility = View.VISIBLE
        } else {
            noDataTextView?.visibility = View.GONE
            addTemplateButtons(view, templates)
            viewTemplatesBtn?.setOnClickListener {
                val intent = Intent(context, DetailsActivity::class.java)
                intent.putExtra("FRAGMENT_ID", 5)
                context?.startActivity(intent)
                Log.d("Button", "View Templates clicked")
            }
        }
    }
}